api/v1alpha3/cloudprovider_encoding.go

```diff
@@ -249,9 +249,8 @@ func isEmpty(val reflect.Value) bool {
 
 // MarshalCloudProviderArgs marshals the cloud provider arguments for passing
 // into a pod spec
-func (c *CPIConfig) MarshalCloudProviderArgs() []string {
-	c.ProviderConfig.Cloud.ExtraArgs = map[string]string{}
-	args := c.ProviderConfig.Cloud.ExtraArgs
+func (cpic *CPIProviderConfig) MarshalCloudProviderArgs() []string {
+	args := cpic.Cloud.ExtraArgs
```

Comment:
if `ExtraArgs` is not what is following will panic

Reply:
stuck it on CPICloudConfig now, where it can't be nil.